src/environment.jl

```diff
@@ -85,6 +100,7 @@ function Init()
     REFCOUNT[] = 1
     atexit(refcount_dec)
 
+    _warn_if_wrong_mpi()
```

Comment:
Needs to be moved to after the `mpi_init_hooks`, as that is when `MPI.COMM_WORLD` is set for Open MPI.

Reply:
Good catch, thanks!